portsyncd/linksync.cpp

```diff
@@ -174,7 +174,7 @@ void LinkSync::onMsg(int nlmsg_type, struct nl_object *obj)
 
     unsigned int flags = rtnl_link_get_flags(link);
     bool admin = flags & IFF_UP;
-    bool oper = flags & IFF_LOWER_UP;
+    bool oper = flags & IFF_RUNNING;
```

Comment:
For e.g, this interface has LOWER_UP but state is UNKNOWN.  Will it have IFF_RUNNING set?

58: Loopback0: <BROADCAST,NOARP,UP,`LOWER_UP`> mtu 65536 qdisc noqueue state `UNKNOWN` mode DEFAULT group default qlen 1000
    link/ether 96:a7:77:d1:73:34 brd ff:ff:ff:ff:ff:ff

Reply:
Yes. Linux command "ip link show" does not show it here. The bgp "show interface" will show this flag. Here is an example of interface info got by linux command and bgp command.

admin@Linecard2:\~$ 
admin@Linecard2:\~$ ```ip link show | grep -A1 Loopback0```
10: Loopback0: <BROADCAST,NOARP,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN mode DEFAULT group default qlen 1000
    link/ether f6:55:7a:eb:85:eb brd ff:ff:ff:ff:ff:ff
admin@Linecard2:\~$ 
admin@Linecard2:\~$ ```docker exec bgp vtysh -c "show interface" | grep -A12 Loopback0```
Interface Loopback0 is ```up```, line protocol is up
  Link ups:       1    last: 2019/03/02 21:32:48.16
  Link downs:     0    last: (never)
  vrf: default
  index 10 metric 0 mtu 65536 speed 0 
  ```flags: <UP,BROADCAST,RUNNING,NOARP>```
  Type: Unknown
  HWaddr: f6:55:7a:eb:85:eb
  inet 10.1.0.1/32 unnumbered
  inet6 fe80::f455:7aff:feeb:85eb/64
  inet6 fc00:10::1/128
  Interface Type Other
  Interface Slave Type None
admin@Linecard2:~$